### PR TITLE
Use rand implementations provided by Distributions.jl v0.17

### LINKED ISSUE #13: Make use of improvements in Distributions.jl v0.17.0
[Distributions.jl](https://github.com/JuliaStats/Distributions.jl) has made some progress towards using generic containers. This renders some functions / fixes provided in [`src/distributions`](https://github.com/JuliaPOMDP/POMDPModelTools.jl/tree/11e22a3c6d538d0e626860514f4d1ad3f13f1d80/src/distributions) obsolete.

### PR DESCRIPTION
**Pull request by lassepe**
This fixes #13. As @MaximeBouton pointed out, we might want to tag a new version if you guys are happy with these changes.

### CODE REVIEW

**Comment by MaximeBouton** [comment]
Looks good, I think we should: 
- tag POMDPModelTools v0.1.4 with Distributions < v0.17
- merge 
- tag POMDPModelTools v0.1.5 with Distributions >= v0.17 
Does that sound correct?
I believe we should do it before Monday when the package registration system will be migrate to Registrator.

File: src/distributions/distributions_jl.jl [hunk 1]
```diff
@@ -2,40 +2,8 @@
...
-
-# for MvNormal - this should be removed once Distributions.jl PR #597 is in a tagged Distributions.jl release
-rand(rng::AbstractRNG, d::MvNormal) = _rand!(rng, d, Vector{eltype(d)}(undef, length(d)))
-
-function _rand!(rng::AbstractRNG, d::MvNormal, x::VecOrMat)
-    Distributions.add!(Distributions.unwhiten!(d.Σ, randn!(rng, x)), d.μ)
-end
+Base.iterate(d::Categorical) = iterate(1:Distributions.ncategories(d))
+Base.iterate(d::Categorical, state) = iterate(1:Distributions.ncategories(d), state)
+Base.length(d::Categorical) = Distributions.ncategories(d)
```
Comment (zsunberg) [inline src/distributions/distributions_jl.jl]: `iterate` is no longer part of the POMDPs.jl interface, and `length` as defined here actually breaks the Distributions.jl concept of `length` being the length of the return type of `rand`, i.e. 1 for Categorical.

That means this whole file should just be deleted.

Comment (lassepe) [reply src/distributions/distributions_jl.jl]: Okay. Did this and squashed commits with previous changes.

**Comment by zsunberg** [comment]
I am going to tag v0.1.4, merge this, and then tag v0.1.5

**Pull request merged**